### PR TITLE
Update README and pyproject.toml to support Python 3.11 and Poetry 1.3.1

### PR DESCRIPTION
**Pull request by jake-kent**

### CODE REVIEW

File: {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.10.1-slim as base
+FROM python:3.11.0-slim as base
```
[the remark this answers is not in the document]
Comment (jake-kent) [reply {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile:1]: I can push commit to roll this line change back, but can't we just bump the version of brotli/whitenoise then?

[the remark this answers is not in the document]
Comment (gone) [reply {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile:1]: can we do 3.11.1, which is newer than .0

File: {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile [hunk 2]
```diff
@@ -37,7 +37,26 @@ WORKDIR app
...
+    SENTRY_DSN="" \
+    REDIS_HOST="" \
+    REDIS_PORT="" \
+    DATABASE_URL="" \
+    ALLOWED_HOSTS="*" \
+    python ./manage.py collectstatic --noinput
+
+# TODO: Set a sensible initial value for workers and threads based on available CPU cores.
+# N.b. Gunicorn timeout must be shorter than any upstream timeouts, e.g. the gateway timeout.
+CMD gunicorn {{ cookiecutter.repo_name }}.config.asgi:application --bind 0.0.0.0:8080 --workers 3 --threads 1 --max-requests 500 --max-requests-jitter 50  --timeout 25  -k uvicorn.workers.UvicornWorker --access-logfile - --preload
```
Comment (gone) [inline {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile:62]: we recently merged in a patch to use a gunicorn config file - can you double check this against the config file?

File: {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (jake-kent) [reply {{cookiecutter.repo_name}}/compose/prod/django/Dockerfile:1]: okay updated

File: scripts/mac_intel_install.sh [hunk 3]
```diff
@@ -64,9 +64,9 @@ echo "${prefix}Reloading zsh"
 source ~/.zshrc
 
 
-echo "${prefix}Attemping to change pyenv version to 3.10.2"
-pyenv install 3.10.2
-pyenv global 3.10.2
+echo "${prefix}Attemping to change pyenv version to 3.11.0"
+pyenv install 3.11.0
```
Comment (gone) [inline scripts/mac_intel_install.sh:68]: sorry to keep picking at this - can we do 3.11.1 here too

Comment (jake-kent) [reply scripts/mac_intel_install.sh:68]: okay updated

**Pull request merged**